sdk/table/azure-table/azure/table/_shared/table_service_client_base.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.table._shared.base_client import parse_query
+from .base_client import StorageAccountHostsMixin
+
+class TableServiceClientBase(StorageAccountHostsMixin):
+    """ :ivar str account_name: Name of the storage account (Cosmos or Azure)"""
+    def __init__(
+            self, account_url,  # type: Any
+            service, # type: str
+            credential=None,  # type: Union[str,TokenCredential]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Create TableServiceClientBase class for sync and async code.
+
+        :param parsed_url:
+            A parsed url to an Azure Storage account.
+        :type account_url: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+        :type credential: Union[str,TokenCredential]
+        :returns: None
+        """
```

Comment:
again, not the point of this PR, but could we move the docstring from the init method into the class? Our team generally places the docstring in the class (although I see no difference with intellisense / rendered documentation between the two)